### DISCUSSION TITLE
implementations of the paper Learning under weight constraints in networks of temporal encoding spiking neurons

### DISCUSSION THREAD

**Discussion by w5688414** [BindsNET/bindsnet#583] [General]
Does anyone implement square cosine encoder for the input neurons of this paper:

```
Learning under weight constraints in networks of temporal encoding
spiking neurons
```

<img width="492" alt="image" src="https://user-images.githubusercontent.com/12107462/189522020-3ed4d657-bad2-417c-83b0-f5ce57bda5a0.png">

**Comment by SimonInParis** [comment]
I never read that. Sounds to me like an engineering solution to map the x values into cos^2(normalized(x) + random phase).
Why not use a simple normalized(x)^y, testing y in the range [0.1...5.0] to see if it fits better? I don't have any argument for the cosine function to perform better than any other mapping function. I understand the benefit of using a mapping function, though.

BTW I tried to play with Iris with very limited success.
But I agree that normalizing this kind of dataset is a good starting point.
Mapping it before converting to Poisson can prove beneficial (Poisson randomization being somehow equivalent to the random cosine phase across the input neurons).
